src/ilasm/CMakeLists.txt

```diff
@@ -19,31 +20,55 @@ set(ILASM_SOURCES
   assembler.cpp
   prebuilt/asmparse.c
 )
-if(WIN32)
-  set_source_files_properties(prebuilt/asmparse.c PROPERTIES COMPILE_FLAGS "/TP")
-endif(WIN32)
+set_source_files_properties( prebuilt/asmparse.c PROPERTIES LANGUAGE CXX )
+
+if(CLR_CMAKE_PLATFORM_UNIX)
+  add_compile_options(-x c++)
```

Comment:
Yes. That's why I initially converted the extension to cpp even though setting property as CXX. These two are required for unix, or clang fails.